src/watchdog/watchmedo.py

```diff
@@ -610,7 +611,10 @@ def auto_restart(args):
 
     # Handle termination signals by raising a semantic exception which will
     # allow us to gracefully unwind and stop the observer
-    termination_signals = {signal.SIGTERM, signal.SIGINT, signal.SIGHUP}
+    termination_signals = {signal.SIGTERM, signal.SIGINT}
+
+    if platform.system() == 'Linux':
```

Comment:
I would go with a specific check:
```suggestion
    if hasattr(signal, "SIGHUP"):
```